### ISSUE TITLE
var-naming[no-role-prefix] triggering for vars when used with "include_role" in 6.16.1

### ISSUE THREAD

**Issue by Rylon** [ansible/ansible-lint#3465]
##### Summary

We have various roles that encapsulate shared functionality. We use `include_role` to include these shared roles in our main roles, and we pass along variables to the included role using `vars`, for example:

```ansible
- name: Include shared role
  include_role:
    name: shared_role
  vars:
    shared_role_parameter: "a value"
```

In Ansible Lint 6.16.1 we're getting tons of errors for `var-naming[no-role-prefix]` I'm assuming because the prefix `shared_role_` doesn't match the name of the role that is including the shared role or something?

The issue is not present in 6.16.0.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 6.16.0 using ansible 2.14.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

As described in the summary, try to include a second role from a first role, and pass role vars along.

##### Desired Behavior

There should be no failure detected in the linter.

##### Actual Behavior

The linter shows `var-naming[no-role-prefix]` errors for each occurrence.

**Comment by drybjed** [comment]
I think that this is a valid false positive which can be handled using `noqa var-naming[no-role-prefix]`. To fix this correctly, `ansible-lint` would need to check the variable against the imported role; I'm not sure if that can be done currently or is possible with current architecture. But a better solution would be welcome, of course.

**Comment by ssbarnea** [comment]
It is true that in this context, the prefix should be determined not by current role, but by included role, so that is a valid bug.
